src/ripple/basics/impl/Time.cpp

```diff
@@ -18,47 +18,66 @@
 //==============================================================================
 
 #include <BeastConfig.h>
-#include <ripple/basics/Time.h>
+#include <ripple/beast/clock/chrono_util.h>
+#include <ripple/basics/chrono.h>
+#include <iomanip>
+#include <sstream>
+#include <tuple>
 
 namespace ripple {
 
-// VFALCO TODO Tidy this up into a RippleTime object
-
-//
-// Time support
-// We have our own epoch.
-//
-
-boost::posix_time::ptime ptEpoch ()
-{
-    return boost::posix_time::ptime (boost::gregorian::date (2000, boost::gregorian::Jan, 1));
-}
-
-int iToSeconds (boost::posix_time::ptime ptWhen)
+static
+std::tuple<int, unsigned, unsigned>
+civil_from_days(int z) noexcept
 {
-    return ptWhen.is_not_a_date_time ()
-           ? -1
-           : (ptWhen - ptEpoch ()).total_seconds ();
+    z += 719468;
+    const int era = (z >= 0 ? z : z - 146096) / 146097;
+    const unsigned doe = static_cast<unsigned>(z - era * 146097);          // [0, 146096]
+    const unsigned yoe = (doe - doe/1460 + doe/36524 - doe/146096) / 365;  // [0, 399]
+    const int y = static_cast<int>(yoe) + era * 400;
+    const unsigned doy = doe - (365*yoe + yoe/4 - yoe/100);                // [0, 365]
+    const unsigned mp = (5*doy + 2)/153;                                   // [0, 11]
+    const unsigned d = doy - (153*mp+2)/5 + 1;                             // [1, 31]
+    const unsigned m = mp + (mp < 10 ? 3 : -9);                            // [1, 12]
+    return std::tuple<int, unsigned, unsigned>(y + (m <= 2), m, d);
```

Comment:
Yeah, and I know I can't give them all good names.  Ridiculously detailed documentation here: http://howardhinnant.github.io/date_algorithms.html#civil_from_days